portfolio/src/main/java/com/google/sps/servlets/AddBlogServlet.java

```diff
@@ -0,0 +1,90 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.MultipartConfig;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.Part;
+
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.FullEntity;
+import com.google.cloud.datastore.KeyFactory;
+import com.google.cloud.storage.Blob;
+import com.google.cloud.storage.BlobId;
+import com.google.cloud.storage.BlobInfo;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+
+import org.jsoup.Jsoup;
+import org.jsoup.safety.Whitelist;
+
+/** Servlet responsible for creating new tasks. */
+@WebServlet("/blog-add")
+@MultipartConfig
+public class AddBlogServlet extends HttpServlet {
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
+    // Sanitize user input to remove HTML tags and JavaScript.
+    String blogTitle = Jsoup.clean(request.getParameter("blog-title"), Whitelist.none());
+    String blogContent = Jsoup.clean(request.getParameter("blog-content"), Whitelist.none());
+
+    long timestamp = System.currentTimeMillis();
+
+    Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
+    KeyFactory keyFactory = datastore.newKeyFactory().setKind("Blog");
+
+    // Get the file chosen by the user.
+    Part filePart = request.getPart("blog-image");
+    String fileName = filePart.getSubmittedFileName();
+    InputStream fileInputStream = filePart.getInputStream();
+
+    // Upload the file and get its URL
+    String uploadedFileUrl = uploadToCloudStorage(fileName, fileInputStream);
+
+     FullEntity taskEntity =
+        Entity.newBuilder(keyFactory.newKey())
+            .set("blogTtile", blogTitle)
+            .set("blogContent", blogContent)
+            .set("Image", uploadedFileUrl)
+            .build();
+    datastore.put(taskEntity);
+
+  }
+  /** Uploads a file to Cloud Storage and returns the uploaded file's URL. */
+  private static String uploadToCloudStorage(String fileName, InputStream fileInputStream) {
+        String projectId = "msaka-sps-spring21";
```

Comment:
Format should look like:
```
String projectId = "msaka-sps-spring21";
String bucketName = "msaka-sps-spring21.appspot.com";
Storage storage =
  StorageOptions.newBuilder().setProjectId(projectId).build().getService();
```

This is because `StorageOptions.newBuilder().setProjectId(projectId).build().getService();` is a continuation of the previous line.